axon-framework/monitoring/tracing.md

```diff
@@ -0,0 +1,535 @@
+| `axon.tracing.attributeProviders.messageType`         | `true`  | Whether to add the message type as a label when handling a message                                     |
+| `axon.tracing.attributeProviders.payloadType`         | `true`  | Whether to add the payload type as a label when handling a message                                     |
+| `axon.tracing.attributeProviders.metadata`            | `true`  | Whether to add the metadata properties as labels when handling a message                               |
+
+#### Manual configuration
+
+The OpenTelemetry support can also be configured using the `Configurer` of Axon Framework to configure
+the `OpenTelemetrySpanFactory`.
+
+```java
+public class AxonConfigurer {
+
+    public void configure(Configurer configurer) {
+        configurer.defaultConfiguration()
+                  .configureSpanFactory(c -> OpenTelemetrySpanFactory.builder().build());
+    }
+}
+```
+
+Note that when not using Spring boot, tracing each message handler invocation is not supported due to a limitation.
+
+## OpenTracing <a id="opentracing"></a>
+
+OpenTracing is deprecated. If necessary, you can
+use [the OpenTracing extension to Axon Framework](../../extensions/tracing.md).
+The functionality of the extension is rather limited compared to the OpenTelemetry integration, so using that if
+possible is preferred.
+
+## Logging <a id="logging"></a>
+
+Sometimes you don't have an APM system available, for instance during local development. It might still be useful to see
+the traces that would be started and finished to obtain insights. For this purpose, the framework provides
+a `LoggingSpanFactory`.
+
+You can configure the `LoggingSpanFactory` in the following ways:
+{% tabs %}
+{% tab title="Spring Boot Bean" %}
+
+```java
+@Configuration
+public class AxonConfiguration {
+
+    @Bean
+    public SpanFactory spanFactory() {
+        return LoggingSpanFactory.INSTANCE;
+    }
+}
+```
+
+{% endtab %}
+{% tab title="Manual" %}
+
+```java
+public class AxonConfigurer {
+
+    public void configure(Configuration configuration) {
+        configuration.configureSpanFactory(c -> LoggingSpanFactory.INSTANCE);
+    }
+}
+```
+
+{% endtab %}
+{% endtabs %}
+
+## Traced components
+
+Axon Framework provides a large range of components that are traced by the configured `SpanFactory`.
+Each component and their added spans are available for reference in this section, with additional information about how
+they should be interpreted.
+
+It's important to note that the availability of these spans is highly dependent on the application configuration. 
+For instance, some components are only used when using Axon Server, or you might have created your own `CommandBus` 
+implementation which does not call the `SpanFactory` API. 
+
+### Commands
+
+The `CommandBus` is instrumented to create spans for both dispatching and handling commands. 
+
+When using the `AxonServerCommandBus`, there will be two handling and dispatch traces 
```

Comment:
This makes me wonder: how would this act with the `DistributedCommandBus`?
It's similar to Axon Server here, as it also uses a `localSegment` (guess who was first, though).
Not something to solve right away, but *definitely* something to look into for 4.7.0.

Reply:
Similarly...have you checked what the behavior is with the `DisruptorCommandBus`? I think we missed that one too :see_no_evil: 
Differently put, something else to look into.

Reply:
Yes, let's put that on the 4.7.0 list :)